doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+- Send 10000 packets for lossy priorities from non-server port (src) to all server ports (dst)
+- Verify that packets are not dropped on src port
+- Verify that packets are not dropped on dst ports
+- Verify that packets are transmitted from from dst ports
+- Send 10000 packets for lossless priorities from non-server port (src) to all server ports (dst)
+- Verify that some packets are dropped on src port, which means that Rx queue is full
+- Verify that some packets are dropped on dst ports, which means that Tx buffer is full
+
+- Teardown:
+  - Stop ARP responder
+
+### Test case # 3 – Asymmetric PFC On Generate PFC frames
+#### Test objective
+Verify that DUT generates PFC frames only on lossless priorities when asymmetric PFC is enabled
+
+#### Used fixtures
+setup, enable_pfc_asym
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+  - Limit maximum bandwith rate on the destination port by setting "1" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+
+- As SAI attributes stores PFC values like a bit vector, calculate bitmask for each PFC mode according to configured "lossless" priorities:
+  - Calculate bitmask for the PFC value
+  - Calculate bitmask for the asymmetric PFC Tx value
+  - Calculate bitmask for the asymmetric PFC Rx value
+
+- Verify asymetric PFC mode is enabled on each server port
+- Get asymmetric PFC Rx value for all server ports
+- Verify asymmetric PFC Rx value for each server port
+- Get asymmetric PFC Tx value for all server ports
+- Verify asymmetric PFC Tx value for all server ports
+
+- Clear all counters for all ports
+- Get lossless priorities
+- Get lossy priorities
+- Get server ports info
+- Get non server port info (Portchannel peers)
+
+- Send 10000 packets for lossless priorities from all server ports (src) to non-server port (dst)
+- Verify that some packets are dropped on src ports, which means that Rx queue is full
+- Verify that PFC frames are generated for lossless priorities
+- Send 10000 packets for lossy priorities from all server ports (src) to non-server port (dst)
+- Verify that PFC frames are not generated for lossy priorities
+
+- Verify PFC value is restored to default
+
+- Teardown:
+  - Restore maximum bandwith rate on the destination port by setting "0" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+   - Stop ARP responder
+
+### Test case # 4 – Asymmetric PFC "On" handle PFC frames on all priorities
+#### Test objective
+Verify that while receiving PFC frames DUT handle PFC frames on all priorities when asymetric mode is enabled
+
+#### Used fixtures
+setup, pfc_storm_runner, enable_pfc_asym
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+
+- Clear all counters for all ports
+- Get lossless priorities
+- Get lossy priorities
+- Get server ports info
+- Get non server port info (Portchannel peers)
+- Send 10000 packets for lossy priorities from non-server port (src) to all server ports (dst)
+- Verify that packets are not dropped on src port
```

Comment:
This step is incorrect, when asymmetric PFC is enabled on a port it should honor PFC frames on all priorities, please fix

Reply:
Asymmetric PFC is enabled only on server ports, but in mentioned steps packets are sending on non-server ports, so incoming packets should not be dropped on non-server ports.